RecoLocalMuon/GEMSegment/plugins/ME0SegFit.cc

```diff
@@ -0,0 +1,515 @@
+// ------------------------- //
+// --> ME0SegFit.cc 
+// Created:  11.05.2015
+// --> Based on CSCSegFit.cc
+// with last mod: 03.02.2015
+// as found in 750pre2 rel
+// ------------------------- //
+
+#include "RecoLocalMuon/GEMSegment/plugins/ME0SegFit.h"
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include "Geometry/GEMGeometry/interface/ME0EtaPartition.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+
+void ME0SegFit::fit(void) {
+  if ( fitdone() ) return; // don't redo fit unnecessarily
+  short n = nhits();
+  switch ( n ) {
+  case 1:
+    edm::LogVerbatim("ME0SegFit") << "[ME0SegFit::fit] - cannot fit just 1 hit!!";
+    break;
+  case 2:
+    fit2();
+    break;
+  case 3:
+  case 4:
+  case 5:
+  case 6:
+    fitlsq();
+    break;
+  default:
+    edm::LogVerbatim("ME0SegFit") << "[ME0SegFit::fit] - cannot fit more than 6 hits!!";
+  }  
+}
+
+void ME0SegFit::fit2(void) {
+
+  // Just join the two points
+  // Equation of straight line between (x1, y1) and (x2, y2) in xy-plane is
+  //       y = mx + c
+  // with m = (y2-y1)/(x2-x1)
+  // and  c = (y1*x2-x2*y1)/(x2-x1)
+
+
+  // 1) Check whether hits are on the same layer
+  // -------------------------------------------
+  ME0SetOfHits::const_iterator ih = hits_.begin();
+
+  ME0DetId d1 = DetId((*ih)->rawId());
+  int il1     = d1.layer();
+  const ME0RecHit& h1 = (**ih);
+  ++ih;    
+  ME0DetId d2 = DetId((*ih)->rawId());
+  int il2     = d2.layer();
+  const ME0RecHit& h2 = (**ih);
+    
+  // Skip if on same layer, but should be impossible :)
+  if (il1 == il2) {
+    edm::LogVerbatim("ME0SegFit") << "[ME0SegFit:fit2] - 2 hits on same layer!!";
+    return;
+  }
+    
+
+  // 2) Global Positions of hit 1 and 2 and
+  //    Local  Positions of hit 1 and 2 w.r.t. reference ME0 Eta Partition 
+  // ---------------------------------------------------------------------
+  const ME0EtaPartition* roll1 = me0etapartition(d1);
+  GlobalPoint h1glopos = roll1->toGlobal(h1.localPosition());
+  const ME0EtaPartition* roll2 = me0etapartition(d2);
+  GlobalPoint h2glopos = roll2->toGlobal(h2.localPosition());
+    
+  // We want hit wrt first me0 eta partition 
+  // ( = reference m00 eta partition) 
+  // (and local z will be != 0)
+  LocalPoint h1pos = refme0etapart()->toLocal(h1glopos);  
+  LocalPoint h2pos = refme0etapart()->toLocal(h2glopos);  
```

Comment:
they are not available ... to have them I would obtain them exactly by this operation, so no gain in precision. Probably only way would be to look at the geometry xml files and see how far away they are spaced one from another. But I would not like to hardcode such numbers here. Do you have some suggestions here @slava77?